### PR TITLE
Better/more TP commands

### PR DESCRIPTION
**Pull request by LoganDark**
This adds:

`/portal tpme <dest>` - Teleports you to some coordinate in your current dimension
`/portal tpme <entity>` - Teleports you to some entity

And new `/portal tp` command:
`/portal tp <entities> <entity>` - Teleports entities to some other entity
`/portal tp <entities> <dest>` - Teleports entities to some coordinate in your current dimension
`/portal tp <entities> <dim> <dest>` - Teleports entities to some coordinate in dim

Includes #229

### CODE REVIEW

File: src/main/java/com/qouteall/immersive_portals/commands/MyCommandServer.java [hunk 1]
```diff
@@ -996,6 +1111,39 @@ private static int placePortalShift(CommandContext<ServerCommandSource> context)
...
+            if (entity instanceof ServerPlayerEntity) {
+                Global.serverTeleportationManager.invokeTpmeCommand(
+                    (ServerPlayerEntity) entity, targetDim, targetPos
+                );
+            } else {
+                if (targetWorld == entity.world) {
+                    entity.refreshPositionAndAngles(targetPos.x, targetPos.y, targetPos.z, entity.yaw, entity.pitch);
+                    entity.setHeadYaw(entity.yaw);
+                } else {
+                    entity.detach();
```
Comment (qouteall) [inline src/main/java/com/qouteall/immersive_portals/commands/MyCommandServer.java]: Can be replaced by ServerTeleportationManager.changeEntityDimension

Comment (LoganDark) [reply src/main/java/com/qouteall/immersive_portals/commands/MyCommandServer.java]: Then I'd have to calculate the eye position rather than setting normal position

Comment (qouteall) [reply src/main/java/com/qouteall/immersive_portals/commands/MyCommandServer.java]: just subtract the eye height

**Comment by qouteall** [comment]
/portal tpme <dest> is the same as /tp @p <dest>?
/portal tpme <entity> Does it allow selecting remote entity?

**Comment by LoganDark** [comment]
> /portal tpme is the same as /tp @p ?

kind of
> /portal tpme Does it allow selecting remote entity?

yes, it uses the same argument type as /kill and friends
